src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,149 @@
+package duke;
+
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.InvalidCommandException;
+
+import java.util.Scanner;
+
+import static duke.Storage.loadFile;
+import static duke.Storage.saveFile;
+import static duke.TaskList.getCommand;
+
+
+public class Duke {
+    private static final TaskList taskList = new TaskList();
+    private static final Scanner in = initialiseInput();
+    private static Command command;
+    private static String input;
+
+    /**
+     * Main entry point of the application.
+     * Initializes the application and starts the interaction with the user.
+     */
+    public static void main(String[] args) {
+        printInitialMsg();
+        loadFile(taskList);
+        do {
+            scanInput();
+            printReaction();
+        } while (canContinue());
+        saveFile(taskList);
+    }
+
+    private static void printInitialMsg() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+    }
+
+    private static void scanInput() {
+        input = takeInput(in);
+        processInput(taskList, input);
+    }
+
+    private static void printReaction() {
+        switch (command) {
+        case ADD:
+            // Fallthrough
```

Comment:
Is there a reason why `ADD`, `DEADLINE` and `EVENT` why they are handled the same as the `TODO` case? It would be good to specify this in the comments, or remove redundant cases.